src/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -14,12 +14,188 @@
+
+    pal::string_t retval;
+    if (!pal::file_exists(deps_json))
+    {
+        trace::verbose(_X("[%s] does not exist"), deps_json.c_str());
+        return retval;
+    }
+
+    pal::ifstream_t file(deps_json);
+    if (!file.good())
+    {
+        trace::verbose(_X("[%s] could not be opened"), deps_json.c_str());
+        return retval;
+    }
+
+    if (skip_utf8_bom(&file))
+    {
+        trace::verbose(_X("UTF-8 BOM skipped while reading [%s]"), deps_json.c_str());
+    }
+
+    try
+    {
+        const auto root = json_value::parse(file);
+        const auto& json = root.as_object();
+        const auto& libraries = json.at(_X("libraries")).as_object();
+
+        pal::string_t prefix = _STRINGIFY(HOST_POLICY_PKG_NAME) + pal::string_t(_X("/"));
+        for (const auto& library : libraries)
+        {
+            if (starts_with(library.first, prefix, false))
+            {
+                retval = library.first.substr(prefix.size());
+                break;
+            }
+        }
+    }
+    catch (const std::exception& je)
+    {
+        pal::string_t jes;
+        (void)pal::utf8_palstring(je.what(), &jes);
+        trace::error(_X("A JSON parsing exception occurred in [%s]: %s"), deps_json.c_str(), jes.c_str());
+    }
+    trace::verbose(_X("%s version is [%s] in deps json file [%s]"), retval.c_str(), deps_json.c_str());
+    return retval;
+}
+
+bool to_hostpolicy_package_path(const pal::string_t& dir, const pal::string_t& version, pal::string_t* candidate)
+{
+    candidate->clear();
+
+    pal::string_t rel_dir = _STRINGIFY(HOST_POLICY_PKG_REL_DIR);
+    if (DIR_SEPARATOR != '/')
+    {
+        replace_char(&rel_dir, '/', DIR_SEPARATOR);
+    }
+
+    pal::string_t path = dir;
+    append_path(&path, _STRINGIFY(HOST_POLICY_PKG_NAME));
+    append_path(&path, version.c_str());
+    append_path(&path, rel_dir.c_str());
+
+    if (!pal::file_exists(path))
+    {
+        return false;
+    }
+
+    *candidate = path;
+    return true;
+}
+
+bool hostpolicy_exists_in_svc(const pal::string_t& version, pal::string_t* resolved_dir)
+{
+    if (version.empty())
+    {
+        return false;
+    }
+
+    pal::string_t svc_dir;
+    pal::get_default_servicing_directory(&svc_dir);
+    append_path(&svc_dir, _X("pkgs"));
+
+    pal::string_t path;
+    if (to_hostpolicy_package_path(svc_dir, version, &path))
+    {
+        resolved_dir->assign(path);
+        trace::verbose(_X("[%s] exists in servicing [%s]"), LIBHOSTPOLICY_NAME, path.c_str());
+        return true;
+    }
+    trace::verbose(_X("[%s] doesn't exist in servicing [%s]"), LIBHOSTPOLICY_NAME, path.c_str());
+    return false;
+}
+
+bool fx_muxer_t::resolve_impl_dir(host_mode_t mode,
+    const pal::string_t& own_dir,
+    const pal::string_t& fx_dir,
+    const pal::string_t& app_candidate,
+    const pal::string_t& deps_file,
+    const std::vector<pal::string_t>& probe_realpaths,
```

Comment:
Since the pal::string objects aren't changing either, should this be ```std::vector<const pal::string_t>``` to be sure nobody changes the underlying strings?